pkg/metrics/metrics.go

```diff
@@ -186,10 +188,19 @@ func New(log logr.Logger, c clock.Clock) *Metrics {
 		)
 	)
 
+	// Create Registry and register the recommended collectors
+	registry := prometheus.NewRegistry()
+	registry.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
+	registry.MustRegister(
+		collectors.NewGoCollector(
+			collectors.WithGoCollectorRuntimeMetrics(collectors.MetricsAll),
+			collectors.WithoutGoCollectorRuntimeMetrics(regexp.MustCompile("^/godebug/.*")),
```

Comment:
Add a comment to the code explaining why we omit the `godebug` metrics.
And give an example of the metrics that will not be available.

OR

Remove that line and include all the Go metrics.

The DefaultRegisterer doesn't omit them.
Nor could I find any examples of other Go processes that omit these.
 * https://github.com/prometheus/client_golang/blob/main/prometheus/registry.go#L60-L62
 * https://github.com/letsencrypt/boulder/blob/5be3650e564e9fbc3492f2ff17678fdf9476efd7/cmd/shell.go#L262-L265
 * https://github.com/istio/istio/blob/3b01bc01992b64c5d7710fbe0e32c74aca925170/pilot/cmd/pilot-agent/status/server.go#L163-L166
 * https://github.com/search?q=repo%3Acilium%2Fcilium%20NewGoCollector&type=code
 * https://github.com/cloudflare/pint/blob/af68e2a13dc46e87688752c4a96d9f81ec78b870/cmd/pint/watch.go#L170-L173

I tried removing that line and observed ~75 extra lines in the  metrics.

```
$ kubectl get --raw /api/v1/namespaces/cert-manager/services/cert-manager:9402/proxy/metrics | wc -l
395

# Vs
$ kubectl get --raw /api/v1/namespaces/cert-manager/services/cert-manager:9402/proxy/metrics  | wc -l
470
```

And 25 extra metrics:

```
$ kubectl get --raw /api/v1/namespaces/cert-manager/services/cert-manager:9402/proxy/metrics  | grep '^go_godebug_'
go_godebug_non_default_behavior_execerrdot_events_total 0
go_godebug_non_default_behavior_gocachehash_events_total 0
go_godebug_non_default_behavior_gocachetest_events_total 0
go_godebug_non_default_behavior_gocacheverify_events_total 0
go_godebug_non_default_behavior_gotypesalias_events_total 0
go_godebug_non_default_behavior_http2client_events_total 0
go_godebug_non_default_behavior_http2server_events_total 0
go_godebug_non_default_behavior_httplaxcontentlength_events_total 0
go_godebug_non_default_behavior_httpmuxgo121_events_total 1
go_godebug_non_default_behavior_installgoroot_events_total 0
go_godebug_non_default_behavior_jstmpllitinterp_events_total 0
go_godebug_non_default_behavior_multipartmaxheaders_events_total 0
go_godebug_non_default_behavior_multipartmaxparts_events_total 0
go_godebug_non_default_behavior_multipathtcp_events_total 0
go_godebug_non_default_behavior_panicnil_events_total 0
go_godebug_non_default_behavior_randautoseed_events_total 0
go_godebug_non_default_behavior_tarinsecurepath_events_total 0
go_godebug_non_default_behavior_tls10server_events_total 0
go_godebug_non_default_behavior_tlsmaxrsasize_events_total 0
go_godebug_non_default_behavior_tlsrsakex_events_total 0
go_godebug_non_default_behavior_tlsunsafeekm_events_total 0
go_godebug_non_default_behavior_x509sha1_events_total 0
go_godebug_non_default_behavior_x509usefallbackroots_events_total 0
go_godebug_non_default_behavior_x509usepolicies_events_total 0
go_godebug_non_default_behavior_zipinsecurepath_events_total 0

$ kubectl get --raw /api/v1/namespaces/cert-manager/services/cert-manager:9402/proxy/metrics  | grep '^go_godebug_' | wc -l
25
```